addons/website_sale/tests/test_website_sequence.py

```diff
@@ -14,7 +14,23 @@ def setUp(self):
         # if stock is installed we can't archive since there is orderpoints
         if hasattr(self.env['product.product'], 'orderpoint_ids'):
             product_templates.mapped('product_variant_ids.orderpoint_ids').write({'active': False})
-        product_templates.write({'active': False})
+
+        # Service on Timesheet can't be archived nor deleted (TASK ID 2221046)
+        # So if the product is service timesheet,
+        # we use the force arg to bypass the rule. Only valid in testing.
+        time_product = None
+        try:
+            time_product = self.env.ref('sale_timesheet_enterprise.fsm_time_product')
+        except:
+            pass # It means that we are not in enterprise, the product doesn't exist
+
+        if time_product:
+            # If the product exist, there is the force arg available for the product templates
+            product_templates.write({'active': False, 'force': True})
```

Comment:
Instead of writing to a "dummy" field (`force`), you can use the context[1].

```python
product_templates.with_context(force_delete=True).write({'active': False})
```
then in the `write` method you can do:
```python
self.env.context.get('force_delete')
```
to retrieve its value :-)

[1] https://www.odoo.com/documentation/master/reference/orm.html#environment

Reply:
Thank you, I finally grasp the use of context with this example :) That's indeed much better !